src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -1902,10 +1912,17 @@ ExpressionAnalysisResult::ExpressionAnalysisResult(
                     ExpressionActions(
                         before_where,
                         ExpressionActionsSettings::fromSettings(context->getSettingsRef())).execute(before_where_sample);
-                    auto & column_elem = before_where_sample.getByName(query.where()->getColumnName());
-                    /// If the filter column is a constant, record it.
-                    if (column_elem.column)
-                        where_constant_filter_description = ConstantFilterDescription(*column_elem.column);
+
+                    /// For creating parameterized view, query parameters are allowed in select
+                    /// As select will be stored without substituting query parameters, we don't want to evaluate the where expression
+                    if (!query_analyzer.query_options.is_create_parameterized_view)
```

Comment:
not necessary, just a suggestion, seems better this way
```suggestion
                    const bool can_evaluate_filter_column = !query_analyzer.query_options.is_create_parameterized_view;
                    if (can_evaluate_filter_column)
```